packages/url_launcher/example/lib/main.dart

```diff
@@ -65,6 +63,20 @@ class _MyHomePageState extends State<MyHomePage> {
     }
   }
 
+  Future<void> _launchUniversalLinkIos(String url) async {
+    try {
+      if (await canLaunch('https://youtube.com')) {
+        await launch('https://youtube.com',
+          forceSafariVC: false, universalLinksOnly: true);
+      }
+    } on Exception {
```

Comment:
I'm assuming you do this as you assume that it failed because the universal link is not supported? but we probably shouldn't re-try if something else causes the error.

The flow feels a little weird anyway (that we need to catch an exception to know whether we can launch a universal only link), is it possible to add some method that tells the app whether the universal link is supported?

Reply:
On IOS, there is no way to tells if the universal link is supported beside actually trying to launch the universal link with the universalLinksOnly flag to be true. I do agree this is a little weird. I will update the Future to return a boolean.